examples/authentication/server/src/token_exchanger.rs

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Utilities for exchanging authorisation codes for identity tokens and validating the resulting
+//! tokens.
+//!
+//! For reference, see:
+//! https://developers.google.com/identity/protocols/oauth2/openid-connect#exchangecode
+
+use jsonwebtoken::{decode, decode_header, Algorithm, DecodingKey, Header, Validation};
+use log::info;
+use serde::{Deserialize, Serialize};
+use std::{error, fmt};
+use url::form_urlencoded;
+
+use crate::http_client;
+
+/// Claims encoded into the identity token
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Claims {
+    aud: String,
+    exp: usize,
+    iat: usize,
+    iss: String,
+    sub: String,
+    email: String,
+    // TODO(#922): Add nonce
+}
+
+#[derive(Deserialize, Debug)]
+struct TokenResponse {
+    expires_in: usize,
+    id_token: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct KeyList {
+    keys: Vec<Key>,
+}
+
+#[derive(Clone, Deserialize, Debug)]
+struct Key {
```

Comment:
Add link to https://auth0.com/docs/tokens/references/jwks-properties or some other resource. Also explain what this represents.

Reply:
Done.